### ISSUE TITLE
Not dropping items when in a PVP battle

### ISSUE THREAD

**Issue by SwitchK1** [GriefPrevention/GriefPrevention#2113]
### Observed Behavior

when in a fight, players do not drop their items after a player dies. meaning they lost the items they had.

### Expected Behavior

it is expected to just drop items.

### Reproduction steps

Hit a player and start killing them.
the other player will no longer be able to drop items etc.
after the player is dead they cannot see their items drop.
The items are basically non-existent and can no longer be obtained again.

### Stack trace or error log

```shell
N/A
```

### Server version

```shell
arclight-1.20.1-1.0.0-SNAPSHOT-1eb0af5
```

### GriefPrevention version

```shell
Enabling GriefPrevention v16.18.1-7-g64b348a
```

### Configuration

```yaml
N/A
```

### Plugin list

```shell
[23:52:24 INFO] [Console]: Plugins (6): ClearLag, MultiplayerSleep, Graves, GriefPrevention, RHLeafDecay, UltimateAirdrops
```

### Running without GriefPrevention

- [ ] I attempted running the server without GriefPrevention installed.
- [ ] The problem does _not_ occur when GriefPrevention is removed from the server.

### Running with only GriefPrevention

- [x] I attempted running only GriefPrevention on the server.
- [x] The issue still occurs when GriefPrevention is the only plugin running.

### Running on a fresh, clean server installation

- [ ] I attempted testing for the issue on a new server.
- [ ] The issue still occurs on a new server.

### Using unmodified client

- [ ] I attempted testing for the issue with the vanilla client.
- [ ] The issue still occurs when using the vanilla client.

### We appreciate you taking the time to fill out a bug report!

- [X] I searched for similar issues before submitting this bug report.

**Comment by RoboMWM** [comment]
Try with just GP installed - could likely be a plugin conflict.

**Comment by SwitchK1** [comment]
just tried it, did not solve it and still dont drop anything.

> Try with just GP installed - could likely be a plugin conflict.

**Comment by QarthO** [comment]
Try using not the beta version of the plugin
Use 16.18.1 and see if that fixes it

I see these as the potential issues: 
- a potential bug on the beta version
- Some conflict with arclight server jar. (Does it work on paper?)
- An issue with graves plugin (Are u sure u actually removed the graves plugin?)
- Do u have the vanilla keep inventory gamerule disabled?
- the player is Using a modded or hacked client
- Remove any data packs, (could be a conflict)

**Comment by SwitchK1** [comment]
- most likely is a bug.

- tried on paper did not solve the issue.

- the graves plugin was 100% removes and still did not solve the issue.

- keep inventory is disabled

- nobody is using a modded or hacked client

- removing data packs also did not solve the issue.

**Comment by QarthO** [comment]
downgrade to the latest stable version, as I cant replicate this issue on v16.18.1
if this fixes it, then we know the issue is a bug on the beta version

**Comment by SwitchK1** [comment]
yep it works, on older version but sadly not on the newer versions.

**Comment by QarthO** [comment]
Well it should be working on the latest release version. Not the latest BETA version

**Comment by RoboMWM** [comment]
@Jikoo have you been able to look at this? I'm gonna be setting up a test server and try to see what's going on myself, but I imagine the cause was from the damage refactoring you did a while back.

**Comment by Jikoo** [comment]
> @Jikoo have you been able to look at this? I'm gonna be setting up a test server and try to see what's going on myself, but I imagine the cause was from the damage refactoring you did a while back.

I've been out of town for a while, no, haven't had time to do much anything. Unlikely, honestly. I touched one listener related to PVP flagging, which is not the same one that affects the actual drops.
This is the listener I moved:
https://github.com/TechFortress/GriefPrevention/blob/ee818cc2f55f7decb44bb0c14cbf2546a80a2204/src/main/java/me/ryanhamshire/GriefPrevention/EntityDamageHandler.java#L796-L831
All it handles is flagging when players take PVP damage, not anything related to the drop process.
